File: youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/pluginconfig/Parse.java
package com.kaltura.playkit.plugins.youbora.pluginconfig;

import java.util.ArrayList;

public class Parse {

    private Boolean parseManifest;
    private Boolean parseCdnNode;
    private Boolean parseCdnSwitchHeader;
    private ArrayList<String> parseCdnNodeList;
    private String parseCdnNameHeader;
    private Integer parseCdnTTL;

    public Boolean getParseManifest() {
        return parseManifest;
    }

    public void setParseManifest(Boolean parseManifest) {
        this.parseManifest = parseManifest;
    }

    public Boolean getParseCdnNode() {
        return parseCdnNode;
    }

    public void setParseCdnNode(Boolean parseCdnNode) {
        this.parseCdnNode = parseCdnNode;
    }

    public Boolean getParseCdnSwitchHeader() {
        return parseCdnSwitchHeader;
    }

    public void setParseCdnSwitchHeader(Boolean parseCdnSwitchHeader) {
        this.parseCdnSwitchHeader = parseCdnSwitchHeader;
    }

    public ArrayList<String> getParseCdnNodeList() {
        return parseCdnNodeList;
    }

    public void setParseCdnNodeList(ArrayList<String> parseCdnNodeList) {
        this.parseCdnNodeList = parseCdnNodeList;
    }

    public String getParseCdnNameHeader() {
        return parseCdnNameHeader;
    }

    public void setParseCdnNameHeader(String parseCdnNameHeader) {
        this.parseCdnNameHeader = parseCdnNameHeader;
    }

    public Integer getParseCdnTTL() {
        return parseCdnTTL;
    }

    public void setParseCdnTTL(Integer parseCdnTTL) {
        this.parseCdnTTL = parseCdnTTL;
    }
}

File: youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/pluginconfig/Parse.kt
package com.kaltura.playkit.plugins.youbora.pluginconfig

import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName

data class Parse(

    @SerializedName(value = "parseManifest", alternate = ["manifest"])
    @JsonAdapter(ParseManifestDeserializer::class)
    var parseManifest: Manifest? = null,

    @SerializedName(value = "parseCdnSwitchHeader", alternate = ["cdnSwitchHeader"])
    var parseCdnSwitchHeader: Boolean? = null,

    @SerializedName(value = "parseCdnNode", alternate = ["cdnNode"])
    @JsonAdapter(CdnNodeDeserializer::class)
    var parseCdnNode: CdnNode? = null, // This is different in iOS

    @kotlin.Deprecated("Use {@link CdnNode} instead")
    @SerializedName(value = "parseCdnNodeList", alternate = ["list"])
    var parseCdnNodeList: ArrayList<String>? = null,

    @SerializedName(value = "parseCdnNameHeader", alternate = ["cdnNameHeader"])
    var parseCdnNameHeader: String? = null,

    @SerializedName(value = "parseNodeHeader", alternate = ["cdnNodeHeader"])
    var parseNodeHeader: String? = null,

    @SerializedName(value = "parseCdnTTL", alternate = ["cdnTTL"])
    var parseCdnTTL: Int? = null)

